### ISSUE TITLE
[5.5.0]: Custom fields don't load in Settings

### ISSUE THREAD

**Issue by maxpdesign** [craftcms/cms#16095]
### Description
Custom fields don't load in the Settings

### Steps to reproduce

1. Open Settings (when allowAdminChanges is enabled
2. Open Fields
3. Fields keep loading

### Expected behavior
I expect that the fields load so I can make changes to their settings

### Actual behavior
Fields table doesn't load, so fields can't be edited

### Craft CMS version

5.5.0

### PHP version

8.3

### Operating system and version

macOs Sonoma 14.3

### Database type and version

MySQL 8.0

### Image driver and version

DDEV

### Installed plugins and versions

"craftcms/aws-s3": "^2.2.1",
"craftcms/ckeditor": "^4.0.3",
"craftcms/cms": "^5.0.0",
"craftcms/feed-me": "^6.0.0",
"craftcms/postmark": "^3.1.0",
"nystudio107/craft-code-field": "^5.0.0",
"nystudio107/craft-emptycoalesce": "^5.0.0",
"nystudio107/craft-minify": "^5.0.0",
"nystudio107/craft-typogrify": "^5.0.0",
"nystudio107/craft-vite": "^5.0.0",
"putyourlightson/craft-elements-panel": "^3.0.0",
"putyourlightson/craft-sherlock": "^5.0.0",
"putyourlightson/craft-sprig": "^3.0.0",
"solspace/craft-freeform": "^5.1.13.1",
"studioespresso/craft-dumper": "^5.0.1",
"verbb/smith": "^3.0.0",
"vlucas/phpdotenv": "^5.0.0"

**Comment by brandonkelly** [comment]
I’m not able to reproduce that. Are you getting any JavaScript errors on the page?

**Comment by kennethormandy** [comment]
@brandonkelly I think I’m seeing this too—I don’t have a very clear picture of what’s going on though.

![Image](https://github.com/user-attachments/assets/8057abd2-24d1-4332-9e5e-f7974474d650)

Here’s the console error I’m getting:

```
Failed to load resource: the server responded with a status of 500 ()
Vuetable.vue:772 load-error ye
loadFailed @ Vuetable.vue:772
Show 1 more frame
Show less
vue.js?v=1731455317:2 TypeError: Cannot read properties of null (reading 'length')
    at a.canReorder (App.vue:951:1)
    at t.get (vue.js?v=1731455317:2:28230)
    at t.evaluate (vue.js?v=1731455317:2:29226)
    at a.canReorder (vue.js?v=1731455317:2:31412)
    at a.init (App.vue:513:1)
    at _n (vue.js?v=1731455317:2:22233)
    at a.r (vue.js?v=1731455317:2:10674)
    at _n (vue.js?v=1731455317:2:22233)
    at t.$emit (vue.js?v=1731455317:2:43975)
    at a.fireEvent (Vuetable.vue:796:12)
wn @ vue.js?v=1731455317:2
```

If I make `getAllLayouts()` return an empty array, it loads the rest of the table again.

This happens with or without `CRAFT_SAFE_MODE=true`.

**Comment by mfell** [comment]
I agree - I’m experiencing the same issue.

In my case, the Hyper plugin needs to be installed for the field list to load correctly, even though I’m not using any Hyper fields and don’t actually need this plugin. Simply having it installed, even in a deactivated state, resolves the issue.

```
Error
Class "verbb\hyper\links\Asset" not found
1. in /var/www/html/vendor/craftcms/cms/src/models/FieldLayout.phpat line 270
261262263264265266267268269270271272273274275276277278279        if (!isset($this->_tabs)) {
            // go through setTabs() so any mandatory fields get added
            $this->setTabs([]);
        }
 
        if (!isset($this->_cardView)) {
            if (!$this->type) {
                $this->setCardView([]);
            } else {
                $this->setCardView($this->type::defaultCardAttributes());
            }
        }
    }
```

**Comment by i-just** [comment]
Thanks all! This will happen if you have a field layout in your database for e.g. a plugin that is no longer required via composer. I raised a PR for it.

**Comment by mfell** [comment]
> Thanks all! This will happen if you have a field layout in your database for e.g. a plugin that is no longer required via composer. I raised a PR for it.

@i-just Just curious:  
I had used Hyper in Craft 4, but deleted all Hyper fields in control panel. After the update to Craft 5 I do not use / need Hyper at all.  
But there could be still some fragments in the database left?

**Comment by i-just** [comment]
> But there could be still some fragments in the database left?

Correct. When you uninstall a plugin that provides custom fields, the fields of that type are not uninstalled. They stay in the control panel (on purpose) and are reported as missing. Some of those fields (and Hyper is one of them) use field layouts, so those stay in the database, too. We will discuss this internally, but regardless of the outcome of that conversation, the changes from the PR should still be applicable.

**Comment by brandonkelly** [comment]
Craft 5.5.0.1 is out with @i-just’s fix (#16097). Thanks for reporting!
